app/main.py

```diff
@@ -1,108 +1,165 @@
-import logging
+import asyncio
 from typing import Any
 
-from fastapi import Depends, Request, FastAPI, HTTPException
+from fastapi import FastAPI, HTTPException, Depends
 from httpx import HTTPStatusError
 from pytimeparse.timeparse import timeparse
+from starlette.requests import Request
 
 from adapter import init_adapter
-from app.config import Settings
+from app.settings import settings
+from app.exceptions import VerificationFailedError
+from app.middleware import RequestReportMiddleware, RequestCacheMiddleware, SignatureCacheMiddleware
 from app.report import init_db
-from app.report.middlewares import CollectRequestData
-from app.report.models import Verify, StatusReport, GatewayInfo
-from app.utils.cache import Cache
-from app.utils.helper import is_data_source_id_allowed, verify_request_from_bandchain, get_band_signature_hash
+from app.report.models import StatusReport, GatewayInfo, VerifyReport, ProviderResponseReport, RequestReport
+from app.utils.cache import LocalCache, RedisCache
+from app.utils.helper import is_data_source_id_allowed, verify_request_from_bandchain
 from app.utils.log_config import init_loggers
 from app.utils.types import VerifyErrorType
 
+# Setup apps
 app = FastAPI()
```

Comment:
do we still use this app?

Reply:
yes, the other apps are mounted to this app